libraries/utilities.js

```diff
@@ -254,6 +232,74 @@ async function getAnchorIdFromObjectFile(folderName) {
 }
 exports.getAnchorIdFromObjectFile = getAnchorIdFromObjectFile;
 
+exports.getTargetIdFromTargetDat = async function getTargetIdFromTargetDat(targetFolderPath) {
+    return new Promise((resolve, reject) => {
+        // unzip the .dat file and read the unique targetId from the config.info file
+        let unzipperDat = new DecompressZip(path.join(targetFolderPath, 'target.dat'));
+
+        unzipperDat.on('error', function (err) {
+            console.error('.dat Unzipper Error', err);
+            reject(err);
+        });
+
+        unzipperDat.on('extract', async function () {
+            let configFilePath = path.join(targetFolderPath, 'config.info');
+            if (await fileExists(configFilePath)) {
+                // read the id stored within the config.info file (it's actually structured as XML)
+                let targetUniqueId = await getTargetIdFromConfigFile(configFilePath);
+                console.log('TODO: cleanup config.info file instead of leaving it in the folder');
+                resolve(targetUniqueId);
+            } else {
+                reject('config.info not found at ' + configFilePath);
+            }
+        });
+
+        unzipperDat.on('progress', function (_fileIndex, _fileCount) {
+            // console.log('Extracted dat file ' + (fileIndex + 1) + ' of ' + fileCount);
+        });
+
+        unzipperDat.extract({
+            path: targetFolderPath,
+            filter: function (file) {
+                return file.type !== 'SymbolicLink' && file.filename.endsWith('info');
+            }
+        });
+    });
+};
+
+async function getTargetIdFromConfigFile(filePath) {
```

Comment:
This function isn't guaranteed to return the correct resultId since xml2js Parser's parseString could call its callback asynchronously (even though it's sync right now, the way it's structured with a callback function could become async at any time).

I'd recommend making another function getTargetIdFromXML() that wraps only the parseString in a classic `new Promise(function (resolve, reject) {})`